addons/website_mail_channel/static/src/snippets/s_channel/options.js

```diff
@@ -14,6 +14,22 @@ options.registry.Channel = options.Class.extend({
     cleanForSave: function () {
         this.$target.addClass('d-none');
     },
+    /**
+     * when a new snippet dropped from d&d menu :
+     * if we have already created channels => select the first one
+     * else => modal prompt (create a new channel)
+     *
+     * @override
+     */
+    onBuilt() {
+        return this._getPublicChannels().then(channels => {
+            if (channels.length) {
+                this.$target.attr("data-id", +channels[0][0]);
+            } else {
+                this.createChannel();
+            }
+        });
+    },
```

Comment:
The onBuilt method cannot return a promise in 13.3, so you cannot do that here yet (see where the onBuilt method is called we don't await the result). Instead of doing it this way, you could call _getPublicChannels in the `start` method (where you can properly await the async part) and save the channels inside a variable and use those here.

Same for the `createChannel` call, it should be awaited but you can't either: instead of calling the method, simulate a click on the create channel button (the hacky way you already saw and replaced with your `enable` method).